mcp.go

```diff
@@ -0,0 +1,889 @@
+package serpent
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// JSONRPC2 message types
+type JSONRPC2Request struct {
+	JSONRPC string          `json:"jsonrpc"`
+	ID      interface{}     `json:"id"`
+	Method  string          `json:"method"`
+	Params  json.RawMessage `json:"params,omitempty"`
+}
```

Comment:
Is there a reason we shouldn't import the types from a different package apart from "a little copying is better than a little dependency"?

Reply:
There is no specific reason. I looked at the MCP spec and just carried the types over.

We could introduce a library for that, but if it doesn't provide any abstraction that makes it easier to work with (e.g., a layer for supporting different spec versions with polyfills), I considered the "copy and be done with it" approach more straightforward.